common/src/main/java/com/flipkart/varadhi/AsyncResult.java

```diff
@@ -0,0 +1,37 @@
+package com.flipkart.varadhi;
+
+public class AsyncResult<T> {
+    private T result;
+    private Throwable cause;
+
+    public static <T> AsyncResult<T> of(T result, Throwable cause) {
```

Comment:
Vert.x [Future](https://vertx.io/docs/apidocs/io/vertx/core/Future.html) has the same capabilities. Are we doing this because we don't want to couple the common module with vertex dependencies?

Reply:
yes, limit Vertx integration and dependencies to only Rest layer.